### PR TITLE
Feature/trade route

### PR DESCRIPTION
**Pull request by SNuseibeh**
This PR includes the UI for the trade feature, mobile & web responsive. 

Note: Backend server is down, so testing the functionality will be difficult.

### CODE REVIEW

**Comment by alex-greenbull** [comment]
Screenshot_httpsarya-crypto-web-git-feature-trade-route-arya-crypto vercel apptrade_001304

1 : remove 0 as the default value on all the inputs
2. hide title if no take profit set, same fr SL and trailing

**Comment by alex-greenbull** [comment]
Missing exchange selector and asset selector

**Comment by alex-greenbull** [comment]
Screenshot_httpsarya-crypto-web-git-feature-trade-route-arya-crypto vercel apptrade_001308
follow figma

**Comment by SNuseibeh** [comment]
> Missing exchange selector and asset selector

the provider selector is a WIP, just having issues with the backend on that matter.

File: src/components/shared/tables/orderHistoryTable.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import { FC, useEffect, useState } from "react";
+import { useMediaQuery } from "react-responsive";
+import { Col, Row } from "../layout/flex";
+import styles from "./assetsTable.module.scss";
+import { useSelector } from "react-redux";
+import { getOpenOrders, getTrade } from "../../../services/redux/tradeSlice";
+import { ThemedContainer } from "../containers/themedContainer";
+import { formatNumber } from "../../../utils/helpers/prices";
+import { XMarkIcon, TrashIcon } from "@heroicons/react/24/outline";
```
Comment (ameer-greenbull) [inline src/components/shared/tables/orderHistoryTable.tsx:9]: remove if not using

File: src/components/shared/trailingPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  onSelect: Function;
+}
+
+export const TrailingPicker: FC<trailingProps> = ({ type, onSelect }) => {
+  const pickeritems = useMemo(() => {
+    return ["Breakeven", "Percentage"];
+  }, []);
+
+  return (
+    <DropdownMenu.Root>
```
Comment (ameer-greenbull) [inline src/components/shared/trailingPicker.tsx:17]:
```suggestion
    <DropdownMenu.Root modal={false}>
```

File: src/pages/trade/index.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      dispatch(addTradables({ assets: _pairs._tradables }));
+      dispatch(setPairs({ pairs: _pairs._pairs }));
+
+      dispatch(setValidations(_validations));
+      dispatch(setOrderType({ orderType: "MARKET" }));
+    })();
+    return () => {
+      dispatch(clearTrade());
+    };
+  }, [symbol, selectedExchange]);
```
Comment (ameer-greenbull) [inline src/pages/trade/index.tsx:72]:
```suggestion
  }, [symbol, selectedExchange, dispatch]);
```

File: src/services/controllers/trade.ts [hunk 4]
```diff
@@ -12,3 +13,48 @@ export const createSwapTrade = async (
...
+
+  return { _tradables, _pairs };
+};
+
+export const getAssetOpenOrders = async (symbol: string, provider: number) => {
+  const { data } = await axiosInstance.get(
+    `trade-engine/orders/open/?provider=${provider}&symbol=${symbol}`
+  );
+
+  console.log("open order data >>> ", data);
```
Comment (ameer-greenbull) [inline src/services/controllers/trade.ts:57]: remove console.log

File: src/components/shared/tables/openOrderTable.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                      </th>
+                      <th className="text-left">{elm.createdAt}</th>
+                    </>
+                  ) : null}
+
+                  <th>
+                    <Row className="justify-start">
+                      <button
+                        className="bg-transparent"
+                        onClick={() => console.log("close")}
```
Comment (ameer-greenbull) [inline src/components/shared/tables/openOrderTable.tsx:110]: why is this console logging onclick?

**Review by ameer-greenbull — CHANGES_REQUESTED** [review]

**Pull request merged**